### PR TITLE
feat(button): add new `gradient-white` and `gradient-grey` buttonTypes

### PR DESCRIPTION
**Pull request by edleeks87**
### Proposed behaviour

Adds new `buttonType` options for `gradient-white` and `gradient-grey` buttons

NB: I've avoided adding much unit test past what's required for coverage as I prefer chromatic to capture the visual stuff here

image
image
image
image
image

### Current behaviour

No support for gradient buttons

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] Unit tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in provided StackBlitz sandbox/Storybook
- [x] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/components/button/button-types.style.ts
Comment (robinzigmond) [inline src/components/button/button-types.style.ts:16]:
```suggestion
const disabledImageStyle = `
```

**suggestion**: typo in const name

File: src/components/button/button.component.tsx
Comment (robinzigmond) [inline src/components/button/button.component.tsx:123]: **comment** (non-blocking): I actually preferred the previous approach of using an object/map to a `switch` statement. But it's obviously something you cared about (since there was no need to change it) and I don't care enough to block the PR over it.

File: src/components/button/button.stories.mdx
Comment (robinzigmond) [inline src/components/button/button.stories.mdx:217]: **comment** (non-blocking): I don't think there was a need to add so many examples of both the new "gradient" types. Personally I think a single example of each, or perhaps 2 (one with and one without an icon) was sufficient - the rest (`disabled` etc) are just combinations of props that consumers would expect to work anyway, we don't need to have a story for every possible combination of props imo!

If this was done for Chromatic then we could move some of these to test stories?

But not a blocker, I don't object to leaving all these in if you tihnk it's best.

File: src/components/button/button-types.style.ts
Comment (edleeks87) [reply src/components/button/button-types.style.ts:16]: 😆 cheers

File: src/components/button/button.component.tsx
Comment (edleeks87) [reply src/components/button/button.component.tsx:123]: Tbh it was just to avoid having to declare the same colour for both gradient variants, I'm happy to revert it back if the second reviewer also prefers the old approach

File: src/components/button/button.stories.mdx
Comment (edleeks87) [reply src/components/button/button.stories.mdx:217]: yeah I just matched the other variants tbh, my thinking is that once we decide on how we want to structure stories going forward they would all match. I also wanted to make it clear that it supports these but not `destructive` but I think I can achieve that by adding some more text to the main stories. Happy to move these to the test file if the other reviewer agrees

File: src/components/button/button.component.tsx
Comment (robinzigmond) [reply src/components/button/button.component.tsx:123]: oh I hadn't even noticed the repetition 🤦 That's definitely a small argument in favour of your change - overall I still slightly prefer the object but as I said I can easily live with it either way, happy to leave it up to you and/or a second reviewer.

File: src/components/button/button-test.stories.tsx
Comment (edleeks87) [reply src/components/button/button-test.stories.tsx:607]: https://github.com/Sage/carbon/blob/d36baf023839693dfe2c4bb74feb774043e1bc9b/src/components/button/button-test.stories.tsx#L16

It's not being excluded

File: src/components/button/button.component.tsx
Comment (edleeks87) [reply src/components/button/button.component.tsx:123]: No worries Tom has pointed out that there's other repetition there as well 🤦, I'll update it based on his suggestion but likely go with `if` statements as it can be shorthanded a bit more that way

```
const iconColor = () => {
    if (buttonType === "primary") {
      return "--colorsActionMajorYang100";
    }

    if (buttonType.includes("gradient")) {
      return "--colorsActionMinorYin090";
    }

    return "--colorsActionMajor500";
  };

File: src/components/button/button-test.stories.tsx
Comment (edleeks87) [reply src/components/button/button-test.stories.tsx:607]: You're probably right, that story likely has no value at all but I'm reluctant to remove it as part of this ticket as I don't have the time to verify if it's used anywhere by QA etc

**Review by tomdavies73 — APPROVED** [review]

**Review by tomdavies73 — DISMISSED** [review]

**Review by robinzigmond — APPROVED** [review]

**Review by tomdavies73 — APPROVED** [review]

**Review by robinzigmond — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Pull request merged**
